aggregator/src/bin/janus_cli.rs

```diff
@@ -168,14 +142,12 @@ fn install_tracing_and_metrics_handlers(config: &CommonConfig) -> Result<()> {
     Ok(())
 }
 
-async fn write_schema(pool: &Pool) -> Result<()> {
+async fn write_schema<C: Clock>(datastore: &Datastore<C>) -> Result<()> {
     info!("Writing database schema");
-    let db_client = pool.get().await.context("couldn't get database client")?;
-    db_client
-        .batch_execute(SCHEMA)
+    datastore
+        .run_tx(|tx| Box::pin(async move { tx.batch_execute(SCHEMA).await }))
```

Comment:
Going through the datastore to set up the schema imposes a hard requirement that we provide a datastore key (or fake datastore key) that won't be used. I'd prefer we use a database connection directly, in the name of ease-of-use when setting up a database, but this isn't strongly held.

Reply:
I agree; we made a purposeful choice to not require a datastore key when writing the schema, let's not regress on that choice.

Reply:
I see your point, but the reason I did it this way was so that I could use `Datastore`'s support for rolling back transactions to implement a dry-run mode for this subcommand. I'm not sure if it'll be worth the effort and ongoing maintenance burden to set up a means of running a DB transaction that we can roll back without using `Datastore`, or making the `Datastore`'s `Crypter` optional. Do we actually ever want to provision a DB schema without access to the datastore key? My expectation is that anyone running this would run `janus_cli create-datastore-key` (to write the key into Kubernetes secrets) and then `janus_cli write-schema` (which would fetch the data store key from k8s secrets, meaning the operator doesn't actually have to provide it).

Maybe what we should do here is collapse `create-datastore-key` and `write-schema` into a single `provision-database` subcommand that would ensure that the datastore is ready to be used. Does it ever make sense to create a datastore key but _not_ write the schema, or vice-versa?

Reply:
We might want to provision a DB without access to the datastore key: in a production environment, the key should be generated in a secure location, i.e. not on an operator workstation, to avoid exposing the key to an untrusted machine. (The general principle is that secrets should be exposed only to trusted machines serving production workloads, on an as-needed basis.)

OTOH, we might decide to do schema deployments (and, eventually, updates) from operator workstations. We might also introduce other operational commands which touch the DB without needing a datastore.

Maybe we need to rethink the way that dry-run is implemented -- the current approach has a few problems already, such as (a) reverting a transaction won't necessarily reveal all of the issues that would occur on commit and (b) it expects that all operations be done in a single transaction since otherwise operations from the first transaction won't be visible in the following transaction. Instead of adding dry-run to the entire datastore, we could instead introduce a CLI-specific trait which provides the necessary datastore operations, with a "real" implementation on `Datastore` and a "dry run" implementation that does nothing (except perhaps print what would be done).